src/cpu.rs

```diff
@@ -527,6 +549,181 @@ impl Cpu {
+                self.registers.l
+            }
+            Operand::M => {
+                let location = self.registers.get_hl() as usize;
+                self.memory[location] = self.memory[location].wrapping_sub(1);
+                self.memory[location]
+            }
+            _ => panic!("INR only accepts registers or a memory location"),
+        };
+        // update flags
+        self.condition_codes.reset_carry();
+        self.condition_codes.set_zero(res);
+        self.condition_codes.set_sign(res);
+        self.condition_codes.set_parity(res);
+        self.condition_codes
+            .set_aux_carry((res.wrapping_add(1) & 0x0F) == 0x00);
+    }
+
+    fn adc(&mut self, val: u8) {
+        let reg_a = self.registers.a;
+        let carry: u8 = if self.condition_codes.carry { 1 } else { 0 };
+        let res = (reg_a as u16)
+            .wrapping_add(val as u16)
+            .wrapping_add(carry as u16);
+        // put result in accumulator
+        self.registers.a = res as u8;
+        // update flags
+        self.condition_codes.set_zero(res as u8);
+        self.condition_codes.set_sign(res as u8);
+        self.condition_codes.set_parity(res as u8);
+        self.condition_codes.set_carry(res > 0xFF);
+        self.condition_codes
+            .set_aux_carry((reg_a & 0x0F) + (val & 0x0F) + (carry & 0x0F) > 0x0F);
+    }
+
+    fn sub(&mut self, val: u8) {
+        let reg_a = self.registers.a;
+        let res: u16 = (reg_a as u16).wrapping_sub(val as u16);
+        // put result in accumulator
+        self.registers.a = res as u8;
+        // update flags
+        self.condition_codes.set_zero(res as u8);
+        self.condition_codes.set_sign(res as u8);
+        self.condition_codes.set_parity(res as u8);
+        self.condition_codes.set_carry(reg_a < val);
+        self.condition_codes
+            .set_aux_carry((reg_a as i8 & 0x0F) - (val as i8 & 0x0F) >= 0);
+    }
+
+    fn sbb(&mut self, val: u8) {
+        let reg_a = self.registers.a;
+        let borrow: u8 = if self.condition_codes.carry { 1 } else { 0 };
+        let res: u16 = (reg_a as u16)
+            .wrapping_sub(val as u16)
+            .wrapping_sub(borrow as u16);
+        // put result in accumulator
+        self.registers.a = res as u8;
+        // update flags
+        self.condition_codes.set_zero(res as u8);
+        self.condition_codes.set_sign(res as u8);
+        self.condition_codes.set_parity(res as u8);
+        self.condition_codes.set_carry(reg_a < val);
+        self.condition_codes
+            .set_aux_carry((reg_a as i8 & 0x0F) - (val as i8 & 0x0F - (borrow as i8)) >= 0);
+    }
+
+    fn adi(&mut self, val: u8) {
+        self.add(val);
+    }
+
+    fn aci(&mut self, val: u8) {
+        self.adc(val);
+    }
+
+    fn sui(&mut self, val: u8) {
+        self.sub(val);
+    }
+
+    fn sbi(&mut self, val: u8) {
+        self.sbb(val);
+    }
+
+    // inx
+    // dcx
+    // dad
```

Comment:
Are these additional instructions that need to be done?

Reply:
Daren wanted some more instructions to work on because he felt his assignments were a bit small comparatively. So he should be doing these 16 bit instructions.